python3Packages.xarray: 2025.10.1 -> 2025.12.0

matshch opened the pull request:
It looks like GitHub has updated the archive of 2025.10.1, and the old one is not on cache.nixos.org, so users cannot get it.

While we are here, we've decided to update it to 2025.12.0.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

matshch committed:
python3Packages.xarray: fix src hash

It looks like GitHub has updated the archive, and the old one is not on
cache.nixos.org, so users cannot get it.

doronbehar commented:
Thanks for the patch. This can go straight to branch `master`. And, speaking of this package, we can simply update it to [v2025.12.0](https://github.com/pydata/xarray/releases/tag/v2025.12.0) which was released last week.

matshch commented:
The issue is that `2025.10.1` is not in master yet, this problem only exists in staging.

doronbehar committed:
python3.pkgs.xarray: 2025.10.1 -> 2025.12.0

Diff: https://github.com/pydata/xarray/compare/v2025.10.1...v2025.12.0

Changelog: https://github.com/pydata/xarray/blob/v2025.12.0/doc/whats-new.rst
